src/arviz_plots/backend/plotly/__init__.py

```diff
@@ -519,30 +591,38 @@ def text(
     size=unset,
     alpha=unset,
     color=unset,
+    rotation=unset,
     vertical_align="middle",
     horizontal_align="center",
     **artist_kws,
 ):
-    """Interface to plotly for adding text to a plot."""
-    artist_kws.setdefault("showlegend", False)
-    # plotly inverts the meaning of alignment with respect to matplotlib and bokeh
-    vertical_align = {"top": "bottom", "bottom": "top"}.get(vertical_align, vertical_align)
-    horizontal_align = {"right": "left", "left": "right"}.get(horizontal_align, horizontal_align)
+    """Interface to Plotly for adding text to a plot."""
+    artist_kws.setdefault("showarrow", False)
+    yanchor_map = {"top": "top", "middle": "middle", "center": "middle", "bottom": "bottom"}
+    yanchor = yanchor_map.get(vertical_align, "middle")
+    font_kwargs = _filter_kwargs({"size": size, "color": color}, artist_kws.pop("font", {}))
+    kwargs = {
+        "textangle": rotation,
+        "opacity": alpha,
+        "yanchor": yanchor,
+        "align": horizontal_align,
+    }
 
-    textfont_artist_kws = artist_kws.pop("textfont", {}).copy()
-    text_kwargs = {"color": color, "size": size}
-    kwargs = {"opacity": alpha}
-    text_object = go.Scatter(
-        x=np.atleast_1d(x),
-        y=np.atleast_1d(y),
-        text=np.vectorize(str_to_plotly_html)(np.atleast_1d(string)),
-        mode="text",
-        textfont=_filter_kwargs(text_kwargs, textfont_artist_kws),
-        textposition=f"{vertical_align} {horizontal_align}",
-        **_filter_kwargs(kwargs, artist_kws),
-    )
-    target.add_trace(text_object)
-    return text_object
+    x_vals = np.atleast_1d(x)
+    y_vals = np.atleast_1d(y)
+    strings = np.atleast_1d(string)
+
+    annotations = []
+    for x_i, y_i, s_i in zip(x_vals, y_vals, strings):
+        ann = target.add_annotation(
+            x=x_i,
+            y=y_i,
+            text=str_to_plotly_html(s_i),
+            font=font_kwargs,
+            **_filter_kwargs(kwargs, artist_kws),
+        )
+        annotations.append(ann)
```

Comment:
I don't think we need to loop here. We were using `atleast_1d` to make it work with .Scatter, but I don't think we use or need (at least for now) sending arrays of text and their x,y positions directly to the backend.

I am not opposed to use `add_annotation` instead of `.Scatter` with `mode="text"` but we need to do extensive testing to make sure it behaves like the old version did. `horizontal_align` differing from the vertical one and using `align` instead of `xanchor` might have differences for example, the docs on `align` are not at all clear to me